wiring/src/spark_wiring_ble.cpp

```diff
@@ -941,6 +967,7 @@ class BleCharacteristicImpl {
     BleOnDataReceivedCallback callback_;
     void* context_;
     static uint16_t defaultUuidCharCount_;
+    std::function<void(const uint8_t*, size_t)> wiringCallback_;
```

Comment:
Since we anyway have to store `std::function` we could simplify things a bit and just `std::bind` the original C function pointer callback to the appropriate `std::function` type and use that universally: https://godbolt.org/z/dcjbYh Same with all the other cases.